Firmware/Marlin_main.cpp

```diff
@@ -2174,7 +2174,7 @@ bool calibrate_z_auto()
 	plan_buffer_line_destinationXYZE(feedrate / 60);
 	st_synchronize();
 	enable_endstops(endstops_enabled);
-	if (PRINTER_TYPE == PRINTER_MK3) {
+	if((PRINTER_TYPE == PRINTER_MK3) || (PRINTER_TYPE == PRINTER_MK3S)){
```

Comment:
I'm not sure if this is correct. For MK3S the `else` statement is run `current_position[Z_AXIS] = Z_MAX_POS + 9.0;`

Reply:
> I'm not sure if this is correct. For MK3S the `else` statement is run `current_position[Z_AXIS] = Z_MAX_POS + 9.0;`
I recently had to hard reset the MK3S 24v to modify the TMC2130.
The X Axis went all the way to the top stop and then the Axis stopped completely on the steel plate. The machine reported failed calibration. If I inserted the PINDA sensor more, it performed the calibration, but the nozzle scratched the heated bed. I started to find out the cause and noticed that MK3S is not defined in the same way as MK3. And the MK3 has the same height XYZ calibration as the MK3S. After adjusting the sensor, I returned it to its original place, and the calibration is without an alarm.

Reply:
> I'm not sure if this is correct. For MK3S the `else` statement is run `current_position[Z_AXIS] = Z_MAX_POS + 9.0;`

I recently had to hard reset the MK3S 24v to modify the TMC2130.
The X Axis went all the way to the top stop and then the Axis stopped completely on the steel plate. The machine reported failed calibration. If I inserted the PINDA sensor more, it performed the calibration, but the nozzle scratched the heated bed. I started to find out the cause and noticed that MK3S is not defined in the same way as MK3. And the MK3 has the same height XYZ calibration as the MK3S. After adjusting the sensor, I returned it to its original place, and the calibration is without an alarm.

Reply:
Any chance you are seeing this issue? https://github.com/prusa3d/Prusa-Firmware/issues/4304 It seems to happen unpredictably. For example I don't see it on my printer. There is a 150mm Z move that should happen but for some it seems to be ignored completely and afterwards the Z-axis will move the nozzle into the bed

Reply:
> Any chance you are seeing this issue? #4304 It seems to happen unpredictably. For example I don't see it on my printer. There is a 150mm Z move that should happen but for some it seems to be ignored completely and afterwards the Z-axis will move the nozzle into the bed


Yes, I saw it, and the comments. maybe that would solve that too. But I can't replicate the same error here.